args/core/async/readonly_rw_spinlock.cpp

```diff
@@ -0,0 +1,20 @@
+#include <core/async/readonly_rw_spinlock.hpp>
+
+namespace args::core::async
+{
+	std::atomic_uint readonly_rw_spinlock::lastId;
+
+	std::unordered_map<uint, int>& readonly_rw_spinlock::localStates()
+	{
+		static thread_local std::unordered_map<uint, int> states;
```

Comment:
considering that states is one of `idle` ,`read` or `write` as far as I understand it, an `std::unordered_map<>` appears overkill for me, consider replacing it with an `std::array<>` maybe ?

Reply:
nvm I see how this works. depending on the number of threads it still might be better to have a different data structure.